ngraph/test/util/engine/interpreter_engine.cpp

```diff
@@ -71,9 +128,64 @@ testing::AssertionResult test::INTERPRETER_Engine::compare_results(const size_t
             break;
         }
 
-        const auto values_match = m_value_comparators.at(element_type);
-
-        comparison_result = values_match(expected_result_constant, result_tensor);
+        switch (element_type)
+        {
+        case element::Type_t::f16:
```

Comment:
I wonder whether we could leverage here `element_type_traits` https://github.com/openvinotoolkit/openvino/blob/431065d46f5f444558d47300b66de8a7c81717ab/ngraph/src/ngraph/type/element_type_traits.hpp#L23-L38
to get rid of all this switch case statements. These `element_type_traits` are used ie. in Constant: https://github.com/openvinotoolkit/openvino/blob/431065d46f5f444558d47300b66de8a7c81717ab/ngraph/src/ngraph/op/constant.hpp#L437-L443

Reply:
The element type checked in this switch statement is known in runtime, unfortunately it's not possible to resolve it with traits which are only usable in compile time.